tools/debugExample.cc

```diff
@@ -0,0 +1,54 @@
+#include "kickcat/Link.h"
+#include "kickcat/protocol.h"
+#include "kickcat/DebugHelpers.h"
+
+#include <iostream>
+
+#ifdef __linux__
+    #include "kickcat/OS/Linux/Socket.h"
+#elif __PikeOS__
+    #include "kickcat/OS/PikeOS/Socket.h"
+#else
+    #error "Unknown platform"
+#endif
+
+
+using namespace kickcat;
+
+int main(int argc, char* argv[])
+{
+    if (argc != 2)
+    {
+        printf("usage: ./test NIC\n");
+        return 1;
+    }
+
+    auto socket = std::make_shared<Socket>();
+    Link link(socket);
+
+    try
+    {
+        socket->open(argv[1], 2ms);
+        socket->setTimeout(1ms);
```

Comment:
You can directly set 1ms at the open call (the 2ms parameter sets the socket's timeout as well)